packages/shylu/shylu_node/hts/src/shylu_hts_impl_def.hpp

```diff
@@ -4044,41 +4045,47 @@ ondiag_solve (const OnDiagTri& t, Sclr* x, const Int ldx, const Int nrhs,
   const Int nthreads = t.nthreads();
   if (nthreads == 1) {
     t.solve(x, ldx, x, ldx, nrhs);
+#ifdef _OPENMP
+#   pragma omp flush
+#endif
     *t_barrier = step;
   } else {
     // Solve T wrk_ = x.
     t.solve(x, ldx, wrk_.data(), t.get_n(), nrhs);
-    { // Wait for the block row MVPs to finish.
-      const Int done = (step << 1);
-      inv_tri_done[tid] = done;
 #ifdef _OPENMP
 #   pragma omp flush
 #endif
+    { // Wait for the block row MVPs to finish.
+      const Int done = (step << 1);
+      inv_tri_done[tid] = done;
       for (Int i = 0; i < nthreads; ++i)
         while (inv_tri_done[i] < done) ;
     }
+#ifdef _OPENMP
```

Comment:
Needed for updating the current thread's view of memory before reading `_wrk.data()` and `t.get_n()`?